### PR TITLE
create condition for coverage reinstated label

### PR DESCRIPTION
**Pull request by yosefserkez**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [ ] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: TT6-185393486

# A brief description of the changes

Current behavior:
Coverage Reinstated indicator is within the enrollment tile.

New behavior:
Coverage Reinstated is a label in header when the enrollment is active and removed from the enrollment tile itself.

# Feature Flag
This lives under
For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:
ENROLLMENT_PLAN_TILE_UPDATE_IS_ENABLED

- [ ] DC
- [x] ME

# Additional Context

The text that is currently rendered in the body of the tile does not technically reflect the current aasm state. It indicates that the enrollment was reinstated at some point.

Based on the other label updates, we are inferring that the client wishes to see the new "Reinstated Enrollment" label (green color) only if the current aasm state would render as a green label as well (auto_renewing, coverage_selected, renewing_coverage_selected).

In other words, we are assuming that the tile should NOT have a green "Reinstated Enrollment" label if the coverage was cancelled, terminated, expired or has an outstanding verification (these are the red, blue, grey or yellow labels).

### CODE REVIEW

**Review by mdkaraman — DISMISSED** [review]
Looks great!  Covers the special case of "Coverage Reinstated" on active enrollments and includes more comprehensive comments.  Awesome job.

**Review by mdkaraman — APPROVED** [review]
Specs have been updated to reflect the new changes.   Thanks for adding the additional context to the description as well.

**Review by erubinst — APPROVED** [review]
Looks good to me.  Updates label to match aasm state and adds comments for clarity
